docs/arch/adr-008-weave-to-flannel-migration.md

```diff
@@ -0,0 +1,39 @@
+# ADR 8: Weave to Flannel Migration
+
+## Context
+
+As proposed in ADR 6, the future CNI of kURL will be Flannel.
+However, almost all existing installations are running Weave.
+
+A migration path must be provided for those instances.
+
+There were two general proposals for a migration path.
+One was to use Multus-CNI to allow using both Flannel and Weave at the same time before stripping out Weave, allowing an upgrade while only taking down one node at once.
+The other was to run a migration that removed Weave from all instances simultaneously before installing Flannel.
+This would be far simpler, not require an additional IP range, and not require Multus to remain in the cluster going forwards.
+It would also require 'stopping the world' and cause downtime.
+
+## Decision
+
+Provide a stop-the-world migration path to move from Weave to Flannel.
+
+## Solution
+
+Create a script to do the following:
```

Comment:
If the script failed at any given point would it be able to roll back? If not perhaps that belongs in the consequences

Reply:
Agreed, rollback seems plausible but it would likely take more engineering effort to setup (or manual intervention)

Reply:
WDYT about we plan to just doc the steps for manual intervention/rollback if required as how debug it instead of try to implement it?

Reply:
Partly because it depends on where in the process things go sideways
probably easy to write the doc anyways though